internal/ethapi/api.go

```diff
@@ -933,6 +934,111 @@ func (s *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rpc.
 	return result, nil
 }
 
+type RequiredBlockState struct {
+	CompactEip1186Proofs []CompactEip1186Proof `json:"compact_eip1186_proofs"` // sorted by address
+	Contracts            []Contract            `json:"contracts"`              // sorted by address
+	AccountNodes         []TrieNode            `json:"account_nodes"`          // sorted by address
+	StorageNodes         []TrieNode            `json:"storage_nodes"`          // sorted by address
+	BlockHashes          []RecentBlockHash     `json:"block_hashes"`           // sorted by address
+}
+
+type CompactEip1186Proof struct {
+	Address       common.Address  `json:"address"`
+	Balance       big.Int         `json:"balance"`
+	CodeHash      common.Hash     `json:"code_hash"`
+	Nonce         uint64          `json:"nonce"`
+	StorageHash   common.Hash     `json:"storage_hash"`
+	AccountProof  []string        `json:"account_proof"`  // sorted: node nearest to root first
+	StorageProofs []StorageResult `json:"storage_proofs"` // sorted
```

Comment:
Note that this deviates from the spec
 https://github.com/ethereum/execution-apis/pull/455/commits/eb9d698ccbbf596163a28222672ca0b87bb9a0c5

Storage nodes and account nodes are unified as a single "trie nodes"